### DISCUSSION TITLE
Are there any plans for the 1.0 version?

### DISCUSSION THREAD

**Discussion by lionskape** [ai/nanostores#70] [General]
Are there any plans for the 1.0 version?
The current version is 0.x. Seems like it is a blocker for production usage

**Comment by ai** [comment]
I do not have specific semantic in the difference between `0.x` and `1.x`.

We are still using in production.

**Reply by lionskape** [reply]
Does this package follow semver?
If so - there are significant differences between 1 and 0 versions (https://semver.org/#spec-item-5).

**Reply by ai** [reply]
SemVer explains only the difference of major and minor updates for `0.x` and `1.x`.

I can't find any rules when and why project should release `1.x`.

One of non-formal rule is “when we are sure in public API and will not change it often”.

**Reply by lionskape** [reply]
But, differences are significant ("anything may be changed at any time" - sounds scary). Most companies will not treat 0.x solution as production-ready and it may be a blocker for adoption.

**Reply by ai** [reply]
Anything can be changed in any libraries even with `10.x` release branch. This is why we have major releases in SemVer.

You should see how the project helps you in migration rather than check versions.

For instance, Nano Stores 0.5 (major release, which can break anything) will still try to support 0.4 API with deprecated warnings which will help in migration.

0.4 APIs will be removed only in 0.6 release.

**Reply by lionskape** [reply]
So, does it means that 0.4 API can be considered stable, and this project just not uses semver to handle releases?

There is a strong restriction for 0.x version in semver for that ( https://semver.org/#spec-item-4 )
> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

**Reply by ai** [reply]
We guarantee stability inside 0.X.x releases.
